index.test.js

```diff
@@ -0,0 +1,189 @@
+const skeletorStaticFileCopier = require('./index');
+
+const srcFilepath1 = 'src/skeletor';
+const destFilepath1 = 'dest/skeletor';
+const content1 = ['content1.txt', 'content2.txt'];
+
+const srcFilepath2 = 'src/skeletor2';
+const destFilepath2 = 'dest/skeletor2';
+const content2 = ['content1.txt'];
+
+jest.mock('path');
+
+beforeEach(() => {
+    require('fs-extra')._setMockSrc({
+        [srcFilepath1]: content1,
+        [srcFilepath2]: content2
+    });
+});
+
+test('run() returns a complete status object', () => {
+    const config = {
+        directories: [
+            {
+                src: srcFilepath1,
+                dest: destFilepath2
+            }
+        ]
+    };
+
+    const expectedResponse = {
+        status: 'complete',
+        message: '1 directory processed'
+    };
+
+    expect.assertions(1);
+    return skeletorStaticFileCopier().run(config)
+        .then(response => {
+            expect(response).toEqual(expectedResponse);
+        });
+});
+
+describe('handles troublesome configurations', () => {
+    test('run() handles malformed config object', () => {
+        const config = {};
+
+        const expectedResponse = {
+            status: 'error',
+            error: 'Config directories is not found or not an array.'
+        };
+
+        const response = skeletorStaticFileCopier().run(config);
+        expect(response).toEqual(expectedResponse);
+    });
+
+    test('throws error if directories is not an array', () => {
+        const expectedResponse = {
+            status: 'error',
+            error: 'Config directories is not found or not an array.'
+        };
+
+        let config = {
+            directories: {}
+        };
+        let response = skeletorStaticFileCopier().run(config);
+        expect(response).toEqual(expectedResponse);
+
+        config = {directories: 'test'};
+        response = skeletorStaticFileCopier().run(config);
+        expect(response).toEqual(expectedResponse);
+
+        config = {directories: 4};
+        response = skeletorStaticFileCopier().run(config);
+        expect(response).toEqual(expectedResponse);
+
+        config = {directories: true};
+        response = skeletorStaticFileCopier().run(config);
+        expect(response).toEqual(expectedResponse);
+    });
```

Comment:
Consider breaking this test up into multiple tests.  A test should test one thing, but this test is testing multiple input/output scenarios